### PR TITLE
Add YOLOv5 finetuning facilities to the app

### PR DESCRIPTION
**Pull request by naisachetti**
A demonstration video on how to use this module can be found at <https://youtu.be/tKdgpavOQ5I>.

### CODE REVIEW

**Comment by solisoares** [comment]
Hi @naisachetti, thanks for contributing, its a great work!

It might take me a bit to go through everything though, to ease the revision can you update your PR description with a summary of **changes/modifications** for the following core files? And also add/update **test cases** for them?
* _video_inference.py
* video.py
* yolo_finetuning.py

About the results in the page "5_📊_Model's_overall_performance.py", I think that an interactive line plot ([Plotly](https://plotly.com/python/line-charts/)) with train and validation loss would increase development analysis. By the time I trained the v1.0.0, I used the wandb integration Yolo gave, and don't remember if we get a way to output all the "good" metrics into a csv, maybe here https://github.com/ultralytics/yolov5/issues/11057#issuecomment-1811983744?

Also, the main concern for the v1.0.0 was _**separation of concerns**_. 
The training is done by technicians or developers, **but used** at the end by therapists and the general public that may not be familiar to the concept of training or any of the technical terms. I feel that the UI for final users is a bit cluttered with this tech stuff.

That's why, for the v1.0.0 I implemented a separated (and quite uncommented and a bit confusing 😅) [training repository](https://github.com/solisoares/therapy-aid-nn). But I can see how an unified tool can help, mainly if the team has both the technician and the therapist.

So, how do you feel about adding a "Developer mode" toggle to the UI where the training and performance can be used and analysed, leaving a "clean" experience when the user enters the app?
